docs/topics/proc_interacting-with-the-circuit-breaker-booster.adoc

```diff
@@ -41,7 +41,8 @@ If the remote endpoint is unavailable, the `name` service responds with an HTTP
 * the `/api/cb-state` endpoint, which returns the state of the Circuit Breaker.
 The state can be:
 ** _open_ : the circuit breaker is preventing requests from reaching the failed service,
-** _closed_: the circuit breaker is allowing requests to reach the service.
+** _closed_: the circuit breaker is allowing requests to reach the service,
+** _half open_: the circuit breaker is allowing a request to reach the service. If the request succeeds, the state of the service is reset to open. If the request fails, the timer is restarted.
```

Comment:
@huntley Nicely written, but I have a few comments:
1. The _open_ state of the CB is the one that prevents traffic to the protected service. Therefore, you should change

> If the request succeeds, the state of the service is reset to open

to something like:

> If the request succeeds, the state of the circuit breaker is reset to _closed_

2. The half-open CB typically allows multiple initial requests to reach the recovering service, rather than just one. I'd say the point is to allow limited traffic to the service, while preventing it from being flooded with requests immediately upon recovery.
3. If any of the limited amount of initial requests fails, the circuit breaker goes into an open state (which also implies that the recovery timeout counter is restarted)

Reply:
You're very much right here, but please see my comment :-)

Reply:
@huntley @Ladicek after following up with @akoniar of Vertx QE, it seems that the Vertx CB booster has a `half-open` state.

I have found some references to it in the [integration tests|https://github.com/openshiftio-vertx-boosters/vertx-circuit-breaker-booster/blob/master/integration-tests/src/test/java/io/openshift/booster/OpenShiftIT.java#L101] and the [HTML frontend|https://github.com/openshiftio-vertx-boosters/vertx-circuit-breaker-booster/blob/master/greeting-service/src/main/resources/webroot/index.html#L139] of the booster.

Reply:
@Ladicek  @akoniar confirms that the `/api/cb-state/` endpoint of the Vertx booster does indeed return a `half-open` state. Thisi specific to the Vertx booster.
@huntley Seems that we need to wrap the section describing the `half-open` state  in a conditional statement, so it only appears in the Vertx guide. Please let me know if you'd like some more info about how you can do it.